ts/session/messages/outgoing/content/data/ChatMessage.ts

```diff
@@ -0,0 +1,156 @@
+import { DataMessage } from './DataMessage';
+import { SignalService } from '../../../../../protobuf';
+import { MessageParams } from '../../Message';
+import { LokiProfile } from '../../../../../types/Message';
+
+interface AttachmentPointer {
+  id?: number;
+  contentType?: string;
+  key?: Uint8Array;
+  size?: number;
+  thumbnail?: Uint8Array;
+  digest?: Uint8Array;
+  filename?: string;
+  flags?: number;
+  width?: number;
+  height?: number;
+  caption?: string;
+  url?: string;
+}
+
+/** Properties of a Preview. */
+interface Preview {
+    url?: string;
+    title?: string;
+    image?: AttachmentPointer;
+}
+
+interface QuotedAttachment {
+  contentType?: string;
+  fileName?: string;
+  thumbnail?: AttachmentPointer;
+}
+
+interface Quote {
+  id?: number;
+  author?: string;
+  text?: string;
+  attachments?: [QuotedAttachment];
```

Comment:
```suggestion
  attachments?: Array<QuotedAttachment>;
```
`[Type]` is a tuple in ts